### PR TITLE
feat(risedev): use docker for kafka service

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See https://github.com/risingwavelabs/risingwave/issues/16903 for the background.

Use docker image `confluentinc/cp-kafka` to start Kafka in RiseDev.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
ping

File: e2e_test/source_inline/commands.toml [hunk 1]
```diff
@@ -53,6 +48,7 @@ dependencies = ["check-kafka"]
 script = """
 #!/usr/bin/env sh
 set -e
+# TODO: no more kafka binaries with docker! 😭 Can we use rpk instead?
```
Comment (xxchan) [inline e2e_test/source_inline/commands.toml:51]: These can be deleted

**Review by xxchan — APPROVED** [review]
LGTM

File: src/risedevtool/src/task/kafka_service.rs [hunk 2]
```diff
@@ -12,108 +12,63 @@
...
-            .join("bin")
-            .join("kafka-server-start.sh"))
+    fn is_user_managed(&self) -> bool {
+        self.user_managed
     }
 
-    fn kafka(&self) -> Result<Command> {
-        Ok(Command::new(self.kafka_path()?))
+    fn image(&self) -> String {
+        "confluentinc/cp-kafka:7.6.1".to_owned()
```
Comment (stdrc) [inline src/risedevtool/src/task/kafka_service.rs:28]: Is it possible to have a place define all the versions of such docker images? Also is it possible to clean old versions of these images if we later upgrade the version?

**Review by stdrc — APPROVED** [review]
Looks much cleaner! Since kafka can not be gracefully killed with ease, seems that we can bring double `risedev d` back again?

File: src/risedevtool/src/task/kafka_service.rs (hunk 2 above)
Comment (BugenZhao) [reply src/risedevtool/src/task/kafka_service.rs:28]: > Is it possible to have a place define all the versions of such docker images?

You reminded me that the image name should be customizable through `risedev.yml`.

> Also is it possible to clean old versions of these images if we later upgrade the version?

We cannot tell if the image is pulled by RiseDev or the user, so let's just leave it to the user.

File: src/risedevtool/src/config_gen/kafka_gen.rs [hunk 3]
```diff
@@ -1,166 +0,0 @@
...
-# The following configuration specifies the time, in milliseconds, that the GroupCoordinator will delay the initial consumer rebalance.
-# The rebalance will be further delayed by the value of group.initial.rebalance.delay.ms as new members join the group, up to a maximum of max.poll.interval.ms.
-# The default value for this is 3 seconds.
-# We override this to 0 here as it makes for a better out-of-the-box experience for development and testing.
-# However, in production environments the default value of 3 seconds is more suitable as this will help to avoid unnecessary, and potentially expensive, rebalances during application startup.
-group.initial.rebalance.delay.ms=0
-"#
-        )
-    }
-}
```
Comment (kwannoel) [inline src/risedevtool/src/config_gen/kafka_gen.rs:166]: Should we still preserve this config?

Comment (xxchan) [reply src/risedevtool/src/config_gen/kafka_gen.rs:166]: These config are just default values

**Pull request merged**